### PR TITLE
Extensions - CLI: --list-extensions command

### PR DESCRIPTION
**Pull request by bryphe**
Implement the `--list-extensions` command, by hooking up into the `ExtensionScanner` and reporting the available extensions

### CODE REVIEW

File: src/Store/ExtensionUtility.re [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+                m("Error discovering user extensions: %s", msg)
+              );
+              None;
+            }
+          }
+        )
+        |> Option.map(p => {
+             Log.infof(m => m("Searching for user extensions in: %s", p));
+             p;
+           })
```
Comment (glennsl) [inline src/Store/ExtensionUtility.re]: I often make a `tap` function that's useful for logging in the middle of expressions:

```reason
let tap = (f, x) => {
  f(x);
  x
};
```
Haven't gotten around to it here yet, but maybe you'd want to add it.

Comment (bryphe) [reply src/Store/ExtensionUtility.re]: Cool idea! Added it in our utilities

**Pull request merged**